WebDriverAgentLib/Categories/XCUIElement+FBScrolling.m

```diff
@@ -185,9 +185,11 @@ - (BOOL)fb_scrollToVisibleWithNormalizedScrollDistance:(CGFloat)normalizedScroll
   }
 
   // Cell is now visible, but it might be only partialy visible, scrolling till whole frame is visible
-  targetCellSnapshot = self.fb_takeSnapshot.fb_parentCellSnapshot;
-  CGVector scrollVector = CGVectorMake(targetCellSnapshot.visibleFrame.size.width - targetCellSnapshot.frame.size.width,
-                                       targetCellSnapshot.visibleFrame.size.height - targetCellSnapshot.frame.size.height
+  targetCellSnapshot = [([self fb_cachedSnapshot] ?: [self fb_takeSnapshot]) fb_parentCellSnapshot];
```

Comment:
Please add comments here with the explanation (basically copy the one from the PR description)